test/suites/snapshots.sh

```diff
@@ -7,35 +7,59 @@ test_snapshots() {
   lxc init testimage foo
 
   lxc snapshot foo
-  [ -d "${LXD_DIR}/snapshots/foo/snap0" ]
+  # FIXME: make this backend agnostic
+  if [ "${LXD_BACKEND}" = "dir" ]; then
+    [ -d "${LXD_DIR}/snapshots/foo/snap0" ]
+  fi
 
   lxc snapshot foo
-  [ -d "${LXD_DIR}/snapshots/foo/snap1" ]
+  # FIXME: make this backend agnostic
+  if [ "${LXD_BACKEND}" = "dir" ]; then
+    [ -d "${LXD_DIR}/snapshots/foo/snap1" ]
+  fi
 
   lxc snapshot foo tester
-  [ -d "${LXD_DIR}/snapshots/foo/tester" ]
+  # FIXME: make this backend agnostic
+  if [ "${LXD_BACKEND}" = "dir" ]; then
+    [ -d "${LXD_DIR}/snapshots/foo/tester" ]
+  fi
 
   lxc copy foo/tester foosnap1
   [ -d "${LXD_DIR}/containers/foosnap1/rootfs" ]
 
   lxc delete foo/snap0
-  [ ! -d "${LXD_DIR}/snapshots/foo/snap0" ]
+  # FIXME: make this backend agnostic
+  if [ "${LXD_BACKEND}" = "dir" ]; then
+    [ ! -d "${LXD_DIR}/snapshots/foo/snap0" ]
+  fi
 
   # no CLI for this, so we use the API directly (rename a snapshot)
   wait_for "${LXD_ADDR}" my_curl -X POST "https://${LXD_ADDR}/1.0/containers/foo/snapshots/tester" -d "{\"name\":\"tester2\"}"
-  [ ! -d "${LXD_DIR}/snapshots/foo/tester" ]
+  # FIXME: make this backend agnostic
+  if [ "${LXD_BACKEND}" = "dir" ]; then
+    [ ! -d "${LXD_DIR}/snapshots/foo/tester" ]
+  fi
 
   lxc move foo/tester2 foo/tester-two
   lxc delete foo/tester-two
-  [ ! -d "${LXD_DIR}/snapshots/foo/tester-two" ]
```

Comment:
No, it doesn't with zfs. zfs doesn't make this directory until you actually start the snapshot (viz. my fix above)